src/Octoshift/Commands/WaitForMigrationCommandBase.cs

```diff
@@ -44,35 +47,81 @@ public async Task Handle(string migrationId, string githubPat = null, bool verbo
-                return;
+                _log.LogInformation($"{GithubPat.GetLogFriendlyName()}: ***");
             }
 
-            if (RepositoryMigrationStatus.IsFailed(state))
+            while (true)
             {
-                _log.LogError($"Migration {migrationId} failed for {repositoryName}");
-                throw new OctoshiftCliException(failureReason);
+                if (RepositoryMigrationStatus.IsSucceeded(state))
+                {
+                    _log.LogSuccess($"Migration {migrationId} succeeded for {repositoryName}");
+                    return;
+                }
+
+                if (RepositoryMigrationStatus.IsFailed(state))
+                {
+                    _log.LogError($"Migration {migrationId} failed for {repositoryName}");
+                    throw new OctoshiftCliException(failureReason);
+                }
+
+                _log.LogInformation($"Migration {migrationId} for {repositoryName} is {state}");
+                _log.LogInformation($"Waiting {WaitIntervalInSeconds} seconds...");
+                await Task.Delay(WaitIntervalInSeconds * 1000);
+
+                (state, repositoryName, failureReason) = await githubApi.GetMigration(migrationId);
             }
+        }
+        else if (migrationId.StartsWith(_orgMigrationIdPrefix))
+        {
+            var state = await githubApi.GetOrganizationMigrationState(migrationId);
+
+            _log.LogInformation($"Waiting for org migration (ID: {migrationId}) to finish...");
 
-            _log.LogInformation($"Migration {migrationId} for {repositoryName} is {state}");
-            _log.LogInformation($"Waiting {WaitIntervalInSeconds} seconds...");
-            await Task.Delay(WaitIntervalInSeconds * 1000);
+            if (githubPat is not null)
+            {
+                _log.LogInformation($"{GithubPat.GetLogFriendlyName()}: ***");
+            }
 
-            (state, repositoryName, failureReason) = await githubApi.GetMigration(migrationId);
+            while (true)
+            {
+                if (OrganizationMigrationStatus.IsSucceeded(state))
+                {
+                    _log.LogSuccess($"Migration {migrationId} succeeded");
+                    return;
+                }
+
+                if (OrganizationMigrationStatus.IsFailed(state))
+                {
+                    _log.LogError($"Migration {migrationId} failed");
+                    return;
+                }
+
+                _log.LogInformation($"Migration {migrationId} is {state}");
+                _log.LogInformation($"Waiting {WaitIntervalInSeconds} seconds...");
+                await Task.Delay(WaitIntervalInSeconds * 1000);
+
+                state = await githubApi.GetOrganizationMigrationState(migrationId);
+            }
+        }
+        else
+        {
+            _log.LogError($"Invalid migration id: {migrationId}");
+            throw new OctoshiftCliException($"Invalid migration id: {migrationId}");
```

Comment:
1. I'd create private method to wait for repo or org migration
2. When throwing `OctoshiftCliException` there's no need to `_log.Error()` since the message is going to get logged anyways. 

```c#
if (!migrationId.StartsWith(REPO_MIGRATION_ID_PREFIX) || !migrationId.StartsWith(ORG_MIGRATION_ID_PREFIX))
{
    throw new OctoshiftCliException($"Invalid migration id: {migrationId}");
}

if (migrationId.StartsWith(REPO_MIGRATION_ID_PREFIX))
{
    await WaitForRepositoryMigration(...);
}
else
{
    await WaitForOrgMigration(...);
}
```